### PR TITLE
Fix crashlog errors in Main.brs

### LINKED ISSUE #1058: Crashes in Main.brs
This bug was discovered via a crash log report on roku.com. line 386 is the worst offender with 50+ crashes a day - more than all the rest added together. Most of the others are 1+ crashes per day

**Software Versions**
Jellyfin Server Version: N/A
Roku Client Version: 1.6.3

**Describe the bug**
App occasionally crashes

**Logs**
`Function main(args As Dynamic) As Void; pkg:/source/Main.brs(230)`
The app is occasionally crashing when it encounters [line 230 of Main.brs](https://github.com/jellyfin/jellyfin-roku/blob/master/source/Main.brs#L230). 

`Function main(args As Dynamic) As Void; pkg:/source/Main.brs(370)`
The app is occasionally crashing when it encounters [line 370 of Main.brs](https://github.com/jellyfin/jellyfin-roku/blob/master/source/Main.brs#L370). 

`Function main(args As Dynamic) As Void; pkg:/source/Main.brs(386)`
The app is occasionally crashing when it encounters [line 386 of Main.brs](https://github.com/jellyfin/jellyfin-roku/blob/master/source/Main.brs#L386). Bug was introduced in #875

`Function main(args As Dynamic) As Void; pkg:/source/Main.brs(404)`
The app is occasionally crashing when it encounters [line 404 of Main.brs](https://github.com/jellyfin/jellyfin-roku/blob/master/source/Main.brs#L404). 

`Function main(args As Dynamic) As Void; pkg:/source/Main.brs(416)`
The app is occasionally crashing when it encounters [line 416 of Main.brs](https://github.com/jellyfin/jellyfin-roku/blob/master/source/Main.brs#L416). 

`Function main(args As Dynamic) As Void; pkg:/source/Main.brs(433)`
The app is occasionally crashing when it encounters [line 433 of Main.brs](https://github.com/jellyfin/jellyfin-roku/blob/master/source/Main.brs#L433).

`Function main(args As Dynamic) As Void; pkg:/source/Main.brs(445)`
The app is occasionally crashing when it encounters [line 445 of Main.brs](https://github.com/jellyfin/jellyfin-roku/blob/master/source/Main.brs#L445). 

`Function main(args As Dynamic) As Void; pkg:/source/Main.brs(468)`
The app is occasionally crashing when it encounters [line 468 of Main.brs](https://github.com/jellyfin/jellyfin-roku/blob/master/source/Main.brs#L468). 

`Function main(args As Dynamic) As Void; pkg:/source/Main.brs(491)`
The app is occasionally crashing when it encounters [line 491of Main.brs](https://github.com/jellyfin/jellyfin-roku/blob/master/source/Main.brs#L491). 

`Function main(args As Dynamic) As Void; pkg:/source/Main.brs(495)`
The app is occasionally crashing when it encounters [line 495 of Main.brs](https://github.com/jellyfin/jellyfin-roku/blob/master/source/Main.brs#L495).

`Function main(args As Dynamic) As Void; pkg:/source/Main.brs(516)`
The app is occasionally crashing when it encounters [line 516 of Main.brs](https://github.com/jellyfin/jellyfin-roku/blob/master/source/Main.brs#L516). 

**Additional context**
A BrightScript crashlog report was recently run on roku.com and these bugs are causing the app to crash 70+ times a day - line 386 being the worst. Logs are not provided. Just the function call and line number where the crash occurred.

Keep in mind these errors are coming from production so the line numbers should be compared to the v1.6.3 tag (or latest version of master. same thing) and not the latest version of unstable

### PR DESCRIPTION
**Pull request by cewert**
Fixes for crashes discovered via roku's crashlog report

**Changes**
- Ensure vars are valid before using
- Converted `blah <> invalid` to `isValid(blah)` when found within the same function as the crash

**Issues**
Fixes #1058

### CODE REVIEW

File: source/Main.brs [hunk 1]
```diff
@@ -257,8 +257,12 @@ sub Main (args as dynamic) as void
             ptr = msg.getData()
             ' ptr is for [row, col] of selected item... but we only have 1 row
             series = msg.getRoSGNode()
-            node = series.seasonData.items[ptr[1]]
-            group = CreateSeasonDetailsGroup(series.itemContent, node)
+            if isValid(ptr) and isValid(series?.seasonData?.items)
```
Comment (1hitsong) [inline source/Main.brs]: We should also ensure the count() on ptr is at least 2 and that the item in position 1 is valid.

File: source/Main.brs [hunk 2]
```diff
@@ -484,7 +493,7 @@ sub Main (args as dynamic) as void
                 ' Exit out of the side panel
                 panel = group.findNode("options")
                 panel.visible = false
-                if group.lastFocus <> invalid
+                if isValid(group?.lastFocus)
```
Comment (1hitsong) [inline source/Main.brs]: If this condition fails because group is invalid, then line 499 will throw an error.

File: source/Main.brs [hunk 3]
```diff
@@ -507,7 +516,7 @@ sub Main (args as dynamic) as void
...
```
Comment (1hitsong) [inline source/Main.brs]: If this condition fails because group is invalid, then line 522 will throw an error.

File: source/Main.brs [hunk 4]
```diff
@@ -530,41 +539,46 @@ sub Main (args as dynamic) as void
...
+                    node.control = "stop"
+
+                    ' If node allows retrying using Transcode Url, give that shot
+                    if isValid(node.retryWithTranscoding) and node.retryWithTranscoding
+                        retryVideo = CreateVideoPlayerGroup(node.Id, invalid, node.audioIndex, true, false)
+                        m.global.sceneManager.callFunc("popScene")
+                        if retryVideo <> invalid
+                            m.global.sceneManager.callFunc("pushScene", retryVideo)
+                        end if
+                    else if isValid(node?.showID)
```
Comment (1hitsong) [inline source/Main.brs]: This is doing the opposite of what it did before. This needs to be `if not isValid`

File: source/Main.brs [hunk 5]
```diff
@@ -530,41 +539,46 @@ sub Main (args as dynamic) as void
...
+                    m.global.sceneManager.callFunc("pushScene", video)
+                    m.global.sceneManager.callFunc("deleteSceneAtIndex", 2)
+                else if node.state = "finished"
+                    node.control = "stop"
+
+                    ' If node allows retrying using Transcode Url, give that shot
+                    if isValid(node.retryWithTranscoding) and node.retryWithTranscoding
+                        retryVideo = CreateVideoPlayerGroup(node.Id, invalid, node.audioIndex, true, false)
+                        m.global.sceneManager.callFunc("popScene")
+                        if retryVideo <> invalid
```
Comment (1hitsong) [inline source/Main.brs]: This can be converted to use isValid()

**Review by 1hitsong — CHANGES_REQUESTED** [review]

**Comment by cewert** [comment]
No longer using optional chaining operators and I've addressed all review comments

**Review by 1hitsong — APPROVED** [review]

**Comment by 1hitsong** [comment]
@cewert Other than merge conflict, looks good.

**Pull request merged**